src/puppetlabs/puppetdb/facts.clj

```diff
@@ -130,26 +130,37 @@
    (nil? data) 4
    (coll? data) 5))
 
+(def large-value-threshold 40)
+
 (defn value->valuemap
   [value]
   (let [type-id (value-type-id value)
-        initial-map {:value_type_id type-id
-                     :value_hash (hash/generic-identity-hash value)
-                     :value_string nil
-                     :value_integer nil
-                     :value_float nil
-                     :value_boolean nil
-                     :value nil}]
+        result {:value_type_id type-id
+                :value_string nil
+                :value_integer nil
+                :value_float nil
+                :value_boolean nil
+                :value nil
+                :large_value_hash nil}]
     (if (nil? value)
-      initial-map
-      (let [value-keyword (case type-id
-                            0 :value_string
-                            1 :value_integer
-                            2 :value_float
-                            3 :value_boolean
-                            5 :value)]
-        (assoc initial-map value-keyword value
-          :value (sutils/munge-jsonb-for-storage value))))))
+      result
+      (let [value-key (case type-id
```

Comment:
And regarding value-key, I think it's likely awkward/ugly/inefficient, but correct -- in any case, I'll fix it.